src/status_im2/common/universal_links.cljs

```diff
@@ -183,6 +176,55 @@
               {:db (dissoc db :universal-links/url)}
               (handle-url url))))
 
+(rf/reg-event-fx
+ :universal-links/generate-profile-url
+ (schema/instrument
+  ::generate-profile-url
+  (fn [{:keys [db] :as _cofx} [{:keys [public-key cb]}]]
+    (let [profile?   (not public-key)
+          ens-name?  (if profile?
+                       (get-in db [:profile/profile :ens-name?])
+                       (get-in db [:contacts/contacts public-key :ens-name]))
+          public-key (if profile? (get-in db [:profile/profile :public-key]) public-key)]
+      {:json-rpc/call
+       [{:method     (if ens-name? "wakuext_shareUserURLWithENS" "wakuext_shareUserURLWithData")
+         :params     [public-key]
+         :on-success #(do (rf/dispatch [:universal-links/save-profile-url public-key %])
+                          (when (fn? cb) (cb)))
+         :on-error   #(log/error "failed to wakuext_shareUserURLWithData"
+                                 {:error      %
+                                  :public-key public-key})}]}))
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args
+     [:schema
+      [:?
+       [:map
+        [:public-key {:optional true} :schema.common/public-key]
+        [:cb {:optional true} fn?]]]]]]
+   [:map
+    [:json-rpc/call :schema.common/general-rpc-call]]]))
+
+(rf/reg-event-fx
+ :universal-links/save-profile-url
+ (schema/instrument
+  ::save-profile-url
+  (fn [{:keys [db]} [public-key url]]
+    (when url
+      {:db
+       (cond-> db
+         (get-in db [:contacts/contacts public-key])
+         (assoc-in [:contacts/contacts public-key :universal-profile-url] url)
+         (= public-key (get-in db [:profile/profile :public-key]))
+         (assoc-in [:profile/profile :universal-profile-url] url))}))
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args
+     [:cat :schema.common/public-key :string]]]
+   [:maybe :map]]))
```

Comment:
It is interesting to note that we can (in the future) automatically reinforce the convention documented in the guidelines (https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#registering-event-handlers) that the map of effects should return either nil or a map of effects of the form `{:db ... :fx ...}`, where no other root keys are allowed.